### ISSUE TITLE
Debugging in dqn_agent.py

### ISSUE THREAD

**Issue by Spiegel-Leser** [tensorflow/agents#679]
As pointed out in [#678](https://github.com/tensorflow/agents/issues/678), I get a "loss is inf or nan" error in
```
dqn_agent.train(experience).loss
```
even though the experience has no inf or nan. I try to get more information by stepping into this line, but in a try block I just pass over to the except:, without try the error gets raised. If I put a breakpoint into

C:\Users\[me]\AppData\Local\Programs\Python\Python39\Lib\site-packages\tf_agents\agents\dqn\dqn_agent.py

and there in

```
 def _train(self, experience, weights):
```

VS2019 warns me that the breakpoint will not be reached.

Can I debug in dqn_agent somehow?

**Comment by sguada** [comment]
Yes, you can enable debug_summaries to get more detailed summaries, and also you can set up breakpoints and debug the code.

**Comment by Spiegel-Leser** [comment]
Using debug_summaries, the error message is:

"Loss is inf or nan : Tensor had NaN values
	 [[node CheckNumerics (defined at C:\Users\Jochen\AppData\Local\Programs\Python\Python39\lib\site-packages\tf_agents\agents\dqn\dqn_agent.py:395) ]] [Op:__inference_train_30748]

Errors may have originated from an input operation.
Input Source operations connected to node CheckNumerics:
 loss/add_1 (defined at C:\Users\Jochen\AppData\Local\Programs\Python\Python39\lib\site-packages\tf_agents\utils\common.py:1367)

Function call stack:
train

  Quelle = C:\Users\Jochen\source\repos\TF-agents\Better_CS_play_initial.py
  Stapelüberwachung:
  File "C:\Users\Jochen\source\repos\TF-agents\....py", line 205, in <module> (Current frame)
    train_loss = agent.train(experience).loss"

Still not possible to step into agent.train or putting a breakpoint there! Any ideas ?

**Comment by Spiegel-Leser** [comment]
Debugging works now, using both debug_summaries in creating the agent and justMyCode = false in the debugging options of VS.
